File: frontend/src/api/dto/gpsServiceConnectionDto.ts
import type { GpsServiceType } from './gpsServiceType.ts'
import type { Instant } from './instant.ts'

/**
 * GPS service connection information
 */
export interface GpsServiceConnectionDto {
  /** Service type identifier */
  serviceType: GpsServiceType
  /** Display name of the service */
  displayName: string
  /** When the service was connected */
  connectedAt: Instant
}

File: backend/src/main/java/fr/pedalons/dto/gps/response/GpsServiceConnectionDto.java
package fr.pedalons.dto.gps.response;

import fr.pedalons.domain.gps.GpsServiceConnection;
import fr.pedalons.dto.validation.ValidateSchema;
import fr.pedalons.enums.GpsServiceType;
import java.time.Instant;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "GPS service connection information")
@ValidateSchema
public record GpsServiceConnectionDto(
    @Schema(description = "Service type identifier", required = true) GpsServiceType serviceType,
    @Schema(description = "Display name of the service", required = true) String displayName,
    @Schema(description = "When the service was connected", required = true) Instant connectedAt) {

  public static GpsServiceConnectionDto from(GpsServiceConnection connection) {
    return new GpsServiceConnectionDto(
        connection.getServiceType(),
        connection.getServiceType().getDisplayName(),
        connection.getConnectedAt());
  }
}
